app/Module/CensusAssistantModule.php

```diff
@@ -299,12 +288,39 @@ public static function formatCensusNote(Note $note) {
 			$data      = Filter::escapeHtml($match[4]);
 			$postamble = Filter::escapeHtml($match[5]);
 
+			// Get the column headers for the census to which this note refers
+			// assumes a soundex match between the fact place name and the specific
+			// census censusDate() function
+			$xref      = $note->getXref();
+			$firstIndi = $note->linkedIndividuals('NOTE')[0];
+			foreach ($firstIndi->getFacts('CENS') as $fact) {
+				if (trim($fact->getAttribute('NOTE'), '@') === $xref) {
+					$date    = strip_tags($fact->getDate()->display(false, '%d %M %Y'));
+					$place   = explode(',', strip_tags($fact->getPlace()->getFullName()));
+					$country = Soundex::daitchMokotoff(array_pop($place));
+					break;
+				}
+			}
+
 			$fmt_headers = array();
-			foreach ($headers as $key => $value) {
-				$fmt_headers[$key] = '<span title="' . Filter::escapeHtml($value) . '">' . $key . '</span>';
+			foreach (Census::allCensusPlaces() as $allCensusesOfPlace) {
+				if (Soundex::compare($country, Soundex::daitchMokotoff($allCensusesOfPlace->censusPlace()))) {
+					foreach ($allCensusesOfPlace->allCensusDates() as $census) {
+						if (strcasecmp($census->censusDate(), $date) === 0) {
```

Comment:
Is there a reason we can't simply write `if ($census->censusDate() == $fact->getAttribute('DATE')`